pkg/manifests/errmapreader.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2019 The Cluster Monitoring Operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package manifests
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+// errMapReader wraps a map[string]string and exposes read methods
+// for values and slices.
+// It tracks the first occured error in errMapReader.err
+// when reading values from the underlying map.
+type errMapReader struct {
+	src map[string]string
+	err error
+}
+
+func NewErrMapReader(src map[string]string) *errMapReader {
+	return &errMapReader{src: src}
+}
+
+// Value returns the value from the underlying map if it is present.
+// If it is not present, an error is generated, errMapReader.err is updated
+// and empty string is returned.
+// If an error already occured in a previous call, empty string is returned.
+func (r *errMapReader) Value(key string) string {
+	if r.err != nil {
+		return ""
+	}
+
+	result, ok := r.src[key]
+	if !ok {
+		r.err = fmt.Errorf("key %s is missing", key)
+		return ""
+	}
+
+	return result
+}
+
+// Slice calls errMapReader.Value(key) and parses it as a string slice in JSON format.
+// It returns the slice as a string joined with commas to be consumed for command line arguments.
+// If an error already occured in a previous call, empty string is returned.
+// If an error occurs during parsing, errMapReader.err is updated
+// and empty string is returned.
+func (r *errMapReader) Slice(key string) string {
```

Comment:
It's quite confusing that this is called `Slice` when it returns a string. Perhaps it would be better to directly return a slice and let the caller do the `strings.Join`. Otherwise, consider renaming this method.

Reply:
ack, I was hesitant to leave the Join in here or not, I can do this at consumer side :+1: